homeassistant/components/electric_kiwi/sensor.py

```diff
@@ -6,26 +6,90 @@
 from datetime import datetime, timedelta
 import logging
 
-from electrickiwi_api.model import Hop
+from electrickiwi_api.model import AccountBalance, Hop
 
 from homeassistant.components.sensor import (
     SensorDeviceClass,
     SensorEntity,
     SensorEntityDescription,
+    SensorStateClass,
 )
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_DOLLAR, PERCENTAGE
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 from homeassistant.util import dt as dt_util
 
-from .const import ATTRIBUTION, DOMAIN
-from .coordinator import ElectricKiwiHOPDataCoordinator
+from .const import ACCOUNT_COORDINATOR, ATTRIBUTION, DOMAIN, HOP_COORDINATOR
+from .coordinator import (
+    ElectricKiwiAccountDataCoordinator,
+    ElectricKiwiHOPDataCoordinator,
+)
 
 _LOGGER = logging.getLogger(DOMAIN)
 
 ATTR_EK_HOP_START = "hop_sensor_start"
 ATTR_EK_HOP_END = "hop_sensor_end"
+ATTR_TOTAL_RUNNING_BALANCE = "total_running_balance"
+ATTR_TOTAL_CURRENT_BALANCE = "total_account_balance"
+ATTR_NEXT_BILLING_DATE = "next_billing_date"
+ATTR_HOP_PERCENTAGE = "hop_percentage"
+
+
+@dataclass(frozen=True)
+class ElectricKiwiAccountRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_func: Callable[[AccountBalance], str | datetime]
```

Comment:
```suggestion
    value_func: Callable[[AccountBalance], float | int | datetime]
```
Return proper native types